### DISCUSSION TITLE
I think I know what this is reffering to... for now

### DISCUSSION THREAD

**Discussion by collinlucke** [helldivers-2/api#51] [General]
[Assignment.properties.progress](https://github.com/helldivers-2/api/blob/9bc8d4b82324cfb453866dc461a4b268057c0327/src/Helldivers-2-API/wwwroot/Helldivers-2-API.json#L1239)

As of this writing the current Major Order is to liberate Tibit, Durgen, and Maia. It's currently being represented in data as:
```
progress: {
  0: 0,
  1: 0,
  2:1
}
```
Right now Maia is the only one of those planets that had been liberated. I guessing that Maia is index 2. I'll know better in a few minutes once Tibit is liberated.

Now what does all of this mean? It means we'll never really know how to represent this with each new order. At least not easily. Thanks for reading.

**Comment by dealloc** [comment]
sounds great! any chance you're open to making a PR to update the documentation regarding this?

**Reply by collinlucke** [reply]
I would love to! However - and please forgive my ignorance - where in the project would I make that change?

**Reply by dealloc** [reply]
no problem! if I'm not mistaken this is in the ArrowHead OpenAPI spec?
in that case you would need to change [this](https://github.com/helldivers-2/api/blob/9bc8d4b82324cfb453866dc461a4b268057c0327/src/Helldivers-2-Models/ArrowHead/Assignment.cs#L9) XML comment here

**Reply by dealloc** [reply]
me (or one of the other maintainers) will guide you through the PR once it's open with any additional tasks that need to be made.

This is also a great reminder for me that I need to update/extend our contributing guides :)

**Comment by xSentry** [comment]
The progress does show if the task is fullfilled or not. (0 = false, 1 = true)
They are mapped by index to the task index.

As you can see in the screenshot:
Progress 0 maps to task 0, 1 to 1 and 2 to 2.
In this case currently task 1 is liberating Estanu, and Estanu is the only currently liberated planet.

![grafik](https://github.com/helldivers-2/api/assets/45976427/b0c49679-5ffc-4f44-9839-33166bea4f6f)

**Reply by dealloc** [reply]
Appreciate the detailed writeup @xSentry once we get around writing out more detailed guides etc I'll definitely refer to this
